starky/src/poseidon_bn128_opt.rs

```diff
@@ -234,10 +235,27 @@ impl Poseidon {
     }
 }
 
+const OUTSIZE:usize = 1;
+impl PoseidonTrait for Poseidon {
+    type BaseField = Fr;
+    fn new() -> Poseidon {
+        Self {}
+    }
+
+    /// Hash function
+    /// init_state would be Fr::zero() initially
+    fn poseidon_hash(&self, inp: &Vec<Fr>, init_state: &[Fr], out: usize) -> Result<Vec<Fr>, String> {
+        let result = self.hash_inner(inp, &init_state[0], OUTSIZE)?;
```

Comment:
The OUTSIZE here  actually is out(function argument), we do not have to define a associated constant.